fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/LuceneQueryComponent.java

```diff
@@ -46,23 +45,31 @@
 @Experimental
 public class LuceneQueryComponent implements QueryComponent, ComponentWithComparison, ComponentWithNoChildren {
     private static final ObjectPlanHash BASE_HASH = new ObjectPlanHash("Lucene-Query");
-    //TODO move to index maintainer but for the sake of dependancies its located here right now
-    public static final String FULL_TEXT_KEY_FIELD = "__full_text_key_field__";
 
+    //MultiFieldSearch determines whether MultiFieldQueryParser or QueryParserBase is used.
+    // QueryParserBase expects the query to contain the fields to be run against and takes a default field
+    // which in our use case is the primary key field.
+    // The MultiFieldQueryParser runs the query against all fields listed and uses AND to join them.
+    // If true and fields not empty we run against all fields listed
+    // If false and fields not empty we run with the QueryParserBase not specifying fields
+    // If true and fields empty we run against all fields in the root expression
+    // If false and fields empty we run against the primary key field.
```

Comment:
Are all these cases actually expected? Or is this just to fill out the contract? That is, why would a caller instantiate multi-fields and not give the fields?